net/shadowsocks/src/opnsense/service/templates/OPNsense/Shadowsocks/shadowsocks.conf

```diff
@@ -5,6 +5,7 @@
     "local_port":{{ OPNsense.shadowsocks.general.localport }},
     "password":"{{ OPNsense.shadowsocks.general.password }}",
     "timeout":60,
+    "mode": "tcp_and_udp",
```

Comment:
Don't you want to make this configurable with a checkbox?

Reply:
I don't know well how checkboxes are handled in opnsense, just found a place in the configs that is responsible for udp traffic within the Shadowsocks tunnel. Also, at the other end of the tunnel you can customize the mode as needed, which makes the opnSense side universal

Reply:
Add an additional field to the model, take "enabled" as example:
https://github.com/opnsense/plugins/blob/520960d435516d0faee307dcadc9a632db2c5fa9/net/shadowsocks/src/opnsense/mvc/app/models/OPNsense/Shadowsocks/General.xml#L6

Add the same field here to make it show up in the GUI:
https://github.com/opnsense/plugins/blob/520960d435516d0faee307dcadc9a632db2c5fa9/net/shadowsocks/src/opnsense/mvc/app/controllers/OPNsense/Shadowsocks/forms/general.xml#L3

And then you would just need to adjust the shadowsocks.conf template, to only make the "mode" appear when the checkbox value is 1.